lib/gems/pending/util/miq_ftp_lib.rb

```diff
@@ -56,7 +56,10 @@ def create_directory_structure(directory_path)
       end
       ftp.chdir(directory)
     end
-    ftp.chdir(pwd)
+  rescue Net::FTPPermError => e
+    _log.info("introspection of directories disabled. skipping create_directory_structure")
```

Comment:
If we skip this where will the dump end up on the server? Is it really better to put it somewhere the user didn't expect rather than fail here?

Reply:
So I had a alternative way of doing this:

```ruby
raise e unless @username.nil? && @password.nil?
```

Which would only attempt to do this on anonymous FTP logins.

* * *

Also, notice my comment below.  The assumption was always to add the file from the logged in directory, and this code still returns to that directory once it is done adding directories.

Reply:
the dump ends up on the server - I've double checked with support that it was going through

Reply:
So I am good with this update.

@carbonin is this clear to you?  Basically, we always return to the `root` dir after creating the directory structure, then pass the full path for the file when doing the upload, which will make sure the file is in the proper place.

I am testing that this is the case in "my test" suite as well (though, maybe I should disable `nlst` on there as well...)